tajo-core/src/main/java/org/apache/tajo/querymaster/QueryMaster.java

```diff
@@ -288,6 +292,39 @@ public void stopQuery(QueryId queryId) {
         return;
       }
 
+      // Create dynamic partitions to CatalogStore by running insert query or CTAS query.
+      Query query = queryMasterTask.getQuery();
```

Comment:
As I mentioned in previous comment, here is not good place for it. QueryCompletedTransition in Query could be a better place.